File: MyMechanicAgenda/app/src/main/java/com/santosdaniel/mymechanicagenda/view/GenericListItem.java
package com.santosdaniel.mymechanicagenda.view;

/**
 * Represents all the information to show an item in the recycle view
 */

public class GenericListItem {

    /**
     * Identifier of the item
     */
    @SuppressWarnings("unused")
    public long id;

    /**
     * Uri to the image of the item
     */
    public String imageUri;

    /**
     * Title of the item
     */
    public String title;

    /**
     * Description of the item
     */
    public String description;

    /**
     * Url with details of the item
     */
    public String detailsUrl;
}

File: MyMechanicAgenda/app/src/main/java/com/santosdaniel/mymechanicagenda/view/viewModel/GenericListItem.kt
package com.santosdaniel.mymechanicagenda.view.viewModel

/**
 * Represents all the information to show an item in the recycle view
 */

data class GenericListItem(

        /**
         * Identifier of the item
         */
        var id: Long = 0,

        /**
         * The lookupKey used
         */
        var lookUpKey: String? = null,

        /**
         * Uri to the image of the item
         */
        var imageUri: String? = null,

        /**
         * Url with details of the item
         */
        var thumbnailUri: String? = null,

        /**
         * Title of the item
         */
        var title: String? = null,

        /**
         * Description of the item
         */
        var description: String? = null


)
